sympy/core/mul.py

```diff
@@ -1268,6 +1264,11 @@ def _eval_is_integer(self):
         # result is not an integer
         if n.is_odd and d.is_even:
             return False
+        if n == d:
+            return True
+        if is_rational is None:
+            if d is S.One:
+                return n.is_integer
```

Comment:
If `d` is `S.One` that does that not mean that `n` is self? This seems like a recipe for infinite recursion.

It's possible that you won't see the infinite recursion in all cases because you are protected by the cache (meaning that `n` is `self` most of the time). I think it's worth trying to hit this line with the cache off to see what happens.

Reply:
> If `d` is `S.One` that does that not mean that `n` is self? This seems like a recipe for infinite recursion.

Ummm right. Yeah it could


>  I think it's worth trying to hit this line with the cache off to see what happens.

I'll check this out

Reply:
I tried!
```python3
sympy/integrals/tests/test_integrals.py[170] .........................................w................
............................w.................................ww..w.................w..................
.........                                                                                          [OK]

____________________________________________ slowest tests ____________________________________________
test_issue_4527 - Took 12.534 seconds
test_principal_value - Took 13.202 seconds
test_issue_17671 - Took 13.584 seconds
test_issue_14709b - Took 21.988 seconds
test_heurisch_option - Took 24.170 seconds
test_issue_15494 - Took 284.756 seconds
====================== tests finished: 164 passed, 6 skipped, in 573.51 seconds =======================
True
>>> Mul(symbols('a', rational=None))
a
>>> Mul(symbols('a', rational=None)).is_integer
>>> fraction(Mul(symbols('a', rational=None)))
(a, 1)
>>> _[1] is S.One
True
>>> Mul(symbols('a', rational=None)).is_rational
>>> 
```
also I tried forcing a recursion couldn't get it

Reply:
I'll try some more

Reply:
`Mul(single arg)` is not a Mul...and you also can't store a `1` as an arg as in `Mul(1, a, evaluate=False)`.

Reply:
You need to verify that some input actually follows this codepath.

Reply:
Yupp they fail 😢

Reply:
( @smichr @oscarbenjamin github seems very laggy of late, I received your comments just now ... )

Reply:
> `Mul(single arg)` is not a Mul...and you also can't store a `1` as an arg as in `Mul(1, a, evaluate=False)`.

Yeah I realised that soon after I posted this 😄